src/components/Rockets.js

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { getRockets, AddReserve, CancelReserve } from '../redux/rockets/rockets';
+import '../App.css';
+
+const Rockets = () => {
+  const dispatch = useDispatch();
+
+  const rocketlist = useSelector((data) => data.rocketsReducer);
+  useEffect(() => {
+    if (rocketlist.length === 0) {
+      dispatch(getRockets());
+    }
+  }, []);
+
+  const ReserveHandler = (id) => {
+    dispatch(AddReserve(id));
+  };
+
+  const CancelReserveHandler = (id) => {
+    dispatch(CancelReserve(id));
+  };
+
+  const rocketItem = rocketlist.map((item) => (
```

Comment:
[OPTIONAL] I think you could destructure `item` when mapping through the array. This way you don't repeat `item.id, item.rocket_name...` each time.